### PR TITLE
ARKAI-52 New authentication manager for AD

### PR DESCRIPTION
**Pull request by david-oc-miller**

### CODE REVIEW

File: vagrant/provisioning/arkcase-ce-external-ldap.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+- hosts: all
```
Comment (david-oc-miller) [inline vagrant/provisioning/arkcase-ce-external-ldap.yml:2]: This play will be used for the MK data center VMs

File: vagrant/provisioning/roles/arkcase-ldap-config/templates/spring-config-ldap-pre-3.3.3.xml [hunk 2]
```diff
@@ -0,0 +1,217 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<beans:beans xmlns="http://www.springframework.org/schema/security" xmlns:beans="http://www.springframework.org/schema/beans"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.springframework.org/schema/beans 
+    http://www.springframework.org/schema/beans/spring-beans-4.1.xsd
+    http://www.springframework.org/schema/security http://www.springframework.org/schema/security/spring-security-3.2.xsd">
```
Comment (david-oc-miller) [inline vagrant/provisioning/roles/arkcase-ldap-config/templates/spring-config-ldap-pre-3.3.3.xml:5]: same as the old template

**Comment by david-oc-miller** [comment]
Merging so I can get this code into develop.  Please review, and I'll resolve your comments in a following PR.

File: vagrant/provisioning/roles/arkcase-app/tasks/main.yml [hunk 3]
```diff
@@ -116,10 +116,14 @@
   register: arkcase_war
   when: arkcase_retrieve_strategy == "sftp"
 
+- name: format config server version
+  set_fact:
+    arkcase_config_server_version_formatted: "{{ '-' ~ arkcase_config_server_version if arkcase_config_server_version != '' else '' }}"
```
Comment (bojmil-armedia) [inline vagrant/provisioning/roles/arkcase-app/tasks/main.yml:121]: I knew the dash would be an issue, but I didn't like the versioning without the dash :)

File: vagrant/provisioning/roles/arkcase-ldap-config/templates/spring-config-ldap-pre-3.3.3.xml (hunk 2 above)
Comment (bojmil-armedia) [reply vagrant/provisioning/roles/arkcase-ldap-config/templates/spring-config-ldap-pre-3.3.3.xml:5]: We would need an openldap/apacheds template. The new AD provider won't work correctly on these LDAP servers.

Comment (david-oc-miller) [reply vagrant/provisioning/roles/arkcase-ldap-config/templates/spring-config-ldap-pre-3.3.3.xml:5]: Let's not add OpenLDAP support to the installer until a customer requires it... given our customer base I would be surprised to find anybody running on OpenLDAP.  For the open source guys, Samba should work well.

**Pull request merged**